linera-execution/tests/test_execution.rs

```diff
@@ -167,42 +133,22 @@ impl UserContract for TestApplication {
     fn execute_message(
         &self,
         context: MessageContext,
-        runtime_sender: ContractRuntimeSender,
+        mut runtime_sender: ContractRuntimeSender,
         _message: Vec<u8>,
     ) -> Result<RawExecutionResult<Vec<u8>>, ExecutionError> {
         // Who we are.
         assert_eq!(context.authenticated_signer, Some(self.owner));
-        let app_id = runtime_sender
-            .send_request(|response_sender| {
-                ContractRequest::Base(BaseRequest::ApplicationId { response_sender })
-            })?
-            .recv()
-            .unwrap();
-
-        runtime_sender
-            .send_request(|response_sender| {
-                ContractRequest::Base(BaseRequest::LockViewUserState { response_sender })
-            })?
-            .recv()
-            .unwrap();
+        let app_id = runtime_sender.application_id()?;
+        runtime_sender.lock()?;
 
         // Call ourselves while the state is locked => not ok.
-        runtime_sender.send_sync_request(|response_sender| {
-            ContractRequest::TryCallApplication {
-                authenticated: true,
-                callee_id: app_id,
-                argument: vec![],
-                forwarded_sessions: vec![],
-                response_sender,
-            }
-        })?;
+        runtime_sender.try_call_application(
+            /* authenticated */ true,
+            app_id,
+            vec![],
+            vec![],
+        )?;
```

Comment:
Not sure if this matters, but compared to the previous code this seems to be missing a call to `unlock`. Is that okay?

```suggestion
        )?;
        runtime_sender.unlock()?;
```

Reply:
I can bring back the internal API but it's missing from the wit API.